src/nxt_unit.c

```diff
@@ -3,12 +3,14 @@
  * Copyright (C) NGINX, Inc.
  */
 
+#define _GNU_SOURCE    /* SCM_CREDENTIALS */
```

Comment:
`_GNU_SOURCE` explicitly defined here and  it was defined before in `nxt_unix.h`, but for Linux only. Please check maybe it is enough?

Reply:
Ok, just removing it doesn't works:

```
-o build/src/nxt_unit.o \
-MMD -MF build/src/nxt_unit.dep -MT build/src/nxt_unit.o \
src/nxt_unit.c
In file included from /usr/include/x86_64-linux-gnu/sys/socket.h:33:0,
                 from /usr/include/netinet/in.h:23,
                 from /usr/include/netdb.h:27,
                 from src/nxt_unix.h:150,
                 from src/nxt_main.h:31,
                 from src/nxt_unit.c:8:
src/nxt_unit.c: In function ‘nxt_unit_run_once’:
src/nxt_socket_msg.h:30:32: error: invalid application of ‘sizeof’ to incomplete type ‘nxt_socket_cred_t {aka struct ucred}’
              CMSG_SPACE(sizeof(nxt_socket_cred_t)))
                                ^
src/nxt_unit.c:3191:30: note: in expansion of macro ‘NXT_OOB_RECV_SIZE’
     u_char               oob[NXT_OOB_RECV_SIZE];
                              ^~~~~~~~~~~~~~~~~
src/nxt_unit.c:3191:26: error: unused variable ‘oob’ [-Werror=unused-variable]
     u_char               oob[NXT_OOB_RECV_SIZE];
                          ^~~
cc1: all warnings being treated as errors
build/Makefile:1432: recipe for target 'build/src/nxt_unit.o' failed
make: *** [build/src/nxt_unit.o] Error 1
```

But the problem was that <stdlib.h> was being included after `<nxt_main.h>`. The macro `_GNU_SOURCE` must be defined before any system header is included because it enables compiler features. Inside `<nxt_main.h>` the `<nxt_unix.h>` is the first header included, then it works for unitd.

Moving the `<stdlib.h>` include here to after `<nxt_main.h>` make it works. Thanks.

Reply:
I removed the `<stdlib.h>` include also, as it apparently is already included somewhere else.